lib/new_relic/agent/method_tracer.rb

```diff
@@ -293,64 +239,63 @@ def derived_class_name
         # @api public
         #
         def add_method_tracer(method_name, metric_name_code = nil, options = {})
+          # We defer prepending the traced method module until add_method_tracer
+          # is called, so that the module is prepended to the correct class.
+          prepend(_nr_traced_method_module)
+
           ::NewRelic::Agent.add_or_defer_method_tracer(self, method_name, metric_name_code, options)
         end
 
         # For tests only because tracers must be removed in reverse-order
         # from when they were added, or else other tracers that were added to the same method
         # may get removed as well.
-        def remove_method_tracer(method_name, metric_name_code) # :nodoc:
+        def remove_method_tracer(method_name) # :nodoc:
           return unless Agent.config[:agent_enabled]
-          if method_defined? "#{_traced_method_name(method_name, metric_name_code)}"
-            alias_method method_name, "#{_untraced_method_name(method_name, metric_name_code)}"
-            undef_method "#{_traced_method_name(method_name, metric_name_code)}"
-            ::NewRelic::Agent.logger.debug("removed method tracer #{method_name} #{metric_name_code}\n")
+          if _nr_traced_method_module.method_defined?(method_name)
+            _nr_traced_method_module.remove_method(method_name)
+            ::NewRelic::Agent.logger.debug("removed method tracer #{method_name}\n")
           else
-            raise "No tracer for '#{metric_name_code}' on method '#{method_name}'"
+            raise "No tracer on method '#{method_name}'"
           end
         end
 
         private
 
-        def _add_method_tracer_now(method_name, metric_name_code, options)
+        def _nr_add_method_tracer_now(method_name, metric_name, options)
           NewRelic::Agent.record_api_supportability_metric(:add_method_tracer)
 
           return unless newrelic_method_exists?(method_name)
-          metric_name_code ||= default_metric_name_code(method_name)
-          return if traced_method_exists?(method_name, metric_name_code)
+          metric_name = _nr_default_metric_name(method_name) if metric_name.to_s.strip.empty?
```

Comment:
How come you changed it from using `||=` to `if metric_name.to_s.strip.empty?`? 
I think ||= might be more performant, although since this is for setting up the method tracer, it only happening on startup, so performance on that scale isn't as big of a deal compared to things that happen during every transaction or span. was curious though.

Reply:
I was thinking of the odd case where someone passes `metric_name` as a string of just whitespace. I didn't specifically test if that would even make it through to the UI, so I just tried avoiding it altogether.

Reply:
Yeah, the performance impacts of something that only happens on agent start up isn't too concerning, I just wanted to ask about it. I think that reasoning makes sense.